oshi-core/src/test/java/oshi/hardware/NetworksTest.java

```diff
@@ -37,17 +37,70 @@
+            assertTrue("NetworkIF bytes received should not be negative", net.getBytesRecv() >= 0);
+            assertTrue("NetworkIF bytes sent should not be negative", net.getBytesSent() >= 0);
+            assertTrue("NetworkIF packets received should not be negative", net.getPacketsRecv() >= 0);
+            assertTrue("NetworkIF packets sent should not be negative", net.getPacketsSent() >= 0);
+            assertTrue("NetworkIF InErrors should not be negative", net.getInErrors() >= 0);
+            assertTrue("NetworkIF out errors should not be negative", net.getOutErrors() >= 0);
+            assertTrue("NetworkIF in drops should not be negative", net.getInDrops() >= 0);
+            assertTrue("NetworkIF collisions should not be negative", net.getCollisions() >= 0);
+            assertTrue("NetworkIF speed should not be negative", net.getSpeed() >= 0);
+            assertTrue("NetworkIF MTU should not be negative", net.getMTU() >= 0);
+            assertTrue("NetworkIF time stamp should be positive", net.getTimeStamp() > 0);
+
+            net.updateAttributes();
+            assertTrue("NetworkIF bytes received after update attr should not be negative", net.getBytesRecv() >= 0);
+            assertTrue("NetworkIF bytes sent after update attr should not be negative", net.getBytesSent() >= 0);
+            assertTrue("NetworkIF packets received after update attr should not be negative",
+                    net.getPacketsRecv() >= 0);
+            assertTrue("NetworkIF packets sent after update attr should not be negative", net.getPacketsSent() >= 0);
+            assertTrue("NetworkIF in errors after update attr should not be negative", net.getInErrors() >= 0);
+            assertTrue("NetworkIF out errors after update attr should not be negative", net.getOutErrors() >= 0);
+            assertTrue("NetworkIF in drops after update attr should not be negative", net.getInDrops() >= 0);
+            assertTrue("NetworkIF collisions after update attr should not be negative", net.getCollisions() >= 0);
+            assertTrue("NetworkIF speed after update attr should not be negative", net.getSpeed() >= 0);
+            assertTrue("NetworkIF time stamp after update attr should not be negative", net.getTimeStamp() > 0);
+
+            if (net.getMacaddr().startsWith("00:00:00") || net.getMacaddr().length() < 8) {
+                assertFalse(net.isKnownVmMacAddr());
+            }
+        }
+    }
+
     /**
-     * Test network interfaces extraction.
+     * Test all network interfaces extraction.
      *
      * @throws IOException
      *             Signals that an I/O exception has occurred.
      */
     @Test
-    public void testNetworkInterfaces() throws IOException {
+    public void testAllNetworkInterfaces() throws IOException {
         SystemInfo si = new SystemInfo();
 
-        for (NetworkIF net : si.getHardware().getNetworkIFs()) {
+        for (NetworkIF net : si.getHardware().getAllNetworkIFs()) {
```

Comment:
Given one list is the superset of the other, it seems like we don't need to repeat all the tests, just do the "all" version for the detail and do at least one test for the "non-local/virtual" version to show the result isn't loopback, etc.